synapse/storage/databases/main/push_rule.py

```diff
@@ -661,15 +683,15 @@ async def set_push_rule_enabled(
 
     def _set_push_rule_enabled_txn(
         self,
-        txn,
-        stream_id,
-        event_stream_ordering,
-        user_id,
-        rule_id,
-        enabled,
-        is_default_rule,
-    ):
-        new_id = self._push_rules_enable_id_gen.get_next()
+        txn: LoggingTransaction,
+        stream_id: int,
+        event_stream_ordering: int,
+        user_id: str,
+        rule_id: str,
+        enabled: bool,
+        is_default_rule: bool,
+    ) -> None:
+        new_id = self._push_rules_enable_id_gen.get_next()  # type: ignore[attr-defined]
```

Comment:
This seems to be defined on the DataStore:

https://github.com/matrix-org/synapse/blob/b76f1a4d5f918def1f643910939b80e9e035e07f/synapse/storage/databases/main/__init__.py#L158-L159

I wonder if we could move this to the PushRuleStore?

Reply:
I have moved this.